core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -410,28 +511,15 @@ void unpublish() {
         pendingOps.decrementAndGet();
     }
 
-    /**
-     * allocate value handle
-     *
-     * @return if chunk is full return -1, otherwise return new handle index
-     */
-    int allocateHandle() {
-        int hi = handleIndex.getAndIncrement();
-        if (hi + 1 > handles.length) {
-            return -1;
-        }
-        handles[hi] = new Handle();
-        return hi;
-    }
-
     int allocateEntryAndKey(K key) {
         int ei = entryIndex.getAndAdd(FIELDS);
         if (ei + FIELDS > entries.length) {
             return -1;
         }
 
         // key and value must be set before linking to the list so it will make sense when reached before put is done
-        setEntryField(ei, OFFSET_HANDLE_INDEX, -1); // set value index to -1, value is init to null
+        // setting the value reference to DELETED_VALUE atomically
+        setEntryFieldLong(ei, OFFSET.VALUE_REFERENCE, DELETED_VALUE);
```

Comment:
Setting value reference to "deleted" straight before the assignment looks weird to me. I think we need DELETED_VALUE and INVALID_ VALUE and to use them accordingly. This would make the code much more understandable. I think I wrote about this already somewhere...